lib/MatchingAlgorithm.cc

```diff
@@ -130,7 +131,7 @@ Matching NProposingMatching::compute_matching() {
     free_list.push(v);
     int pref_list_size = v->get_preference_list().size();
     int residual = v->get_upper_quota();
-    bookkeep_data[v] = VertexBookkeeping(0, pref_list_size, residual);
+    bookkeep_data[v] = VertexBookkeeping(0, pref_list_size, 0, residual);
```

Comment:
After a constructor which takes begin, end and residual arguments is added, this change should be reverted.